src/widgets/index/index.ts

```diff
@@ -347,28 +347,28 @@ const index = (widgetParams: IndexWidgetParams): IndexWidget => {
       );
 
       if (localInstantSearchInstance && Boolean(widgets.length)) {
-        const nextState = widgets.reduce((state, widget) => {
-          // the `dispose` method exists at this point we already assert it
-          const next = widget.dispose!({
+        widgets.forEach((widget) => {
+          // doing nothing with the return value of dispose, this is handled by the next part
+          widget.dispose!({
             helper: helper!,
-            state,
+            state: helper!.state,
```

Comment:
if we're actually doing this, dispose should no longer accept helper, state, nor should it return search parameters, but I didn't do that in this PR out of simplicity.